### DISCUSSION TITLE
802.1AR: Can we dress a plan of features related to this standard?

### DISCUSSION THREAD

**Discussion by raskyld** [netbox-community/netbox#14298] [Ideas]
Hello there!

I am new to the Netbox community, I've read your contribution guideline, and the excellent https://github.com/bradfitz/issue-tracker-behaviors and hope I am actually complying with it by opening a discussion instead of an Issue right now!

Quick introduction, I've been involved mostly with Software Engineering but I've, since a couple of months, a growing interest in cybersecurity that I pursue out of business hours and TL;DR I started by the layers I am used the less with and started to take a look at the [IEEE 802.1AR-2018](https://ieeexplore.ieee.org/document/8423794) (available for free through their IEEE GET program) standards and the [document](https://trustedcomputinggroup.org/resource/tpm-2-0-keys-for-device-identity-and-attestation/) that the _Trusted Computing Group_ issued about how to use TPM 2.0 to implement what the standards calls a _DevID Module_.

I would like to do two 🐦 , one 🪨 by continuing my research while contributing to this awesome project.

I open this discussion to see if there is any folks in the _Stadium_ that would be interested to dress a plan with me about how to make Netbox "802.1AR-aware" and also I would greatly appreciate collaborating with someone that could help me perfect my knowledge of Network Engineering.

**Comment by raskyld** [comment]
So I think the first step would be to add a field in the data model that describes whether the associated device is 802.1AR compliant.

**Reply by candlerb** [reply]
You can do that today with a [Custom Field](https://docs.netbox.dev/en/stable/customization/custom-fields/). You could also store additional information such as DevIDs this way.

For more sophisticated modelling which requires its own tables, you can write a [Plugin](https://docs.netbox.dev/en/stable/plugins/development/).

**Reply by raskyld** [reply]
Thanks for the answer!
As I am pretty new to the project I wasn't sure if I should go for a plugin. But I thought because it's a standards it would make sense to implement awareness of this standard in the core of Netbox, how do you go about deciding what goes in the core vs. extensibility mechanisms (custom fields, plugins, ...)
